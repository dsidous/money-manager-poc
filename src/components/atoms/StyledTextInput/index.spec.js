import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { StyledTextInput } from './index';

configure({ adapter: new Adapter() });

describe('styled text input', () => {

  it('renders as expected', () => {
    const render = shallow(<StyledTextInput />);
    expect(render).toMatchSnapshot();
  });

  it('renders as expected obfuscated password', () => {
    const render = shallow(<StyledTextInput password />);
    expect(render).toMatchSnapshot();
  });

});
