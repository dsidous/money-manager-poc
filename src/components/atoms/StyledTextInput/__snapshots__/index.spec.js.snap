// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled text input renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledTextInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoCapitalize": undefined,
      "autoCorrect": undefined,
      "onChangeText": undefined,
      "placeholder": undefined,
      "returnKeyType": undefined,
      "secureTextEntry": undefined,
      "underlineColorAndroid": "rgba(0,0,0,0)",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoCapitalize": undefined,
        "autoCorrect": undefined,
        "onChangeText": undefined,
        "placeholder": undefined,
        "returnKeyType": undefined,
        "secureTextEntry": undefined,
        "underlineColorAndroid": "rgba(0,0,0,0)",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`styled text input renders as expected obfuscated password 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledTextInput
    password={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoCapitalize": undefined,
      "autoCorrect": undefined,
      "onChangeText": undefined,
      "placeholder": undefined,
      "returnKeyType": undefined,
      "secureTextEntry": undefined,
      "underlineColorAndroid": "rgba(0,0,0,0)",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoCapitalize": undefined,
        "autoCorrect": undefined,
        "onChangeText": undefined,
        "placeholder": undefined,
        "returnKeyType": undefined,
        "secureTextEntry": undefined,
        "underlineColorAndroid": "rgba(0,0,0,0)",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
